src/coreclr/vm/util.cpp

```diff
@@ -1953,11 +1953,12 @@ HRESULT GetFileVersion(                     // S_OK or error
 
 Volatile<double> NormalizedTimer::s_frequency = -1.0;
 
-void FillStubCodePage(BYTE* pageBase, const void* code, SIZE_T codeSize, SIZE_T pageSize)
+void FillStubCodePage(BYTE* pageBase, const void* code, SIZE_T codeSize, SIZE_T pageSize, SIZE_T paddingSize = 0)
 {
     SIZE_T totalCodeSize = (pageSize / codeSize) * codeSize;
-
-    memcpy(pageBase, code, codeSize);
+    SIZE_T actualSize = codeSize - paddingSize;
+    memcpy(pageBase, code, actualSize);
+    memset(pageBase + actualSize, 0, paddingSize);
```

Comment:
Zeros are not the right padding to use for all architectures. The padding should use the architecture specific nop or breakpoint instructions. The disassembly may look weird otherwise, and it may also cause odd performance issues as the processor speculative execution logic tries to decode the potentially non-sensical instructions.